Experiment_Launcher_code/ExperimentLauncher.py

```diff
@@ -6,27 +15,34 @@ def main():
     experiment_gui = ExperimentGUI()
     experiment_gui.setup_gui()
 
-    # After the GUI is closed, get the settings using the get_settings method
-    settings = experiment_gui.on_start_clicked()
-
-    if settings:
-        # Extract the necessary parameters from the settings
-        experiment_name = settings.get('experiment_name')
-        comport_name = settings.get('comport_name')
-        num_trials = settings.get('num_trials')
-        return_time = settings.get('return_time')
-        decision_time = settings.get('decision_time')
-        opponent_type = settings.get('opponent_type')
-        opponent1_strategy = settings.get('opponent1_strategy')
-        opponent2_strategy = settings.get('opponent2_strategy')
-        opponent1_probability = settings.get('opponent1_probability')
-        opponent2_probability = settings.get('opponent2_probability')
+    # After the GUI is closed, get the settings using the appropriate methods
+    if experiment_gui.is_data_valid():
+        comport_name = experiment_gui.get_com_port()
+        experiment_parameters = experiment_gui.get_experiment_parameters()
+        opponent_configuration = experiment_gui.get_opponent_configuration()
+
+        # Instantiate software components
+        video_analyzer = Video_Analyzer("1780")
```

Comment:
i dont understand why you want to initialize the video analyser  and reward manager here.seemed better in experiment_manager

Reply:
I agree with a) b) and c) .It makes sense to creat ethe agents in the launcher but I feel like we might be making the PrisonerABC class unnecesarily complex if we add the setVideoAnalyzer() and setRewardManager() methods as they are only use by the mouse monitor and not the fixed strategy class or the learners

Reply:
So, basically you agree with (d) as well :-). 

If we are creating the agents in the launcher and we want the video analyzer and reward manager to be set for the MouseMonitor through its __init__() function  (which is unique to the MouseMonitor and does not require a change in PrisonerABC) we need to instantiate the two SW components in Launcher as well - which is what I did. Also since the ExperimentManager is also using both of them it is set for it as well.

Reply:
yes i think this change sounds good